ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+// MARK: - Drawer Model
+
+@objc public enum DrawerDirection: Int, CaseIterable {
+    /// Drawer originates from left
+    case left
+    /// Drawer originates from right
+    case right
+}
+
+/// `DrawerState` assist to configure drawer functional properties via UIKit components.
+@objc(DrawerState)
+public class DrawerState: NSObject, ObservableObject {
+
+    /// A callback executed when the drawer is expanded/collapsed
+    public var onStateChange: (DrawerStateChangedCompletionBlock)?
+
+    /// Set `isExpanded` to `true` to maximize the drawer's width to fill the device screen horizontally minus the safe areas.
+    /// Set to `false` to restore it to the normal size.
+    @objc @Published public var isExpanded: Bool = false {
+        didSet {
+            onStateChange?(isExpanded)
+        }
+    }
+    @objc @Published public var presentationDirection: DrawerDirection = .left
+
+    /// Set `backgroundDimmed` to `true` to dim the spacer area between drawer and base view.
+    /// If set to `false` it restores to `clear` color
+    @objc @Published public var backgroundDimmed: Bool = false
+}
+
+public typealias DrawerStateChangedCompletionBlock = (_ isExpanded: Bool) -> Void
```

Comment:
Calling this a completion block seems very "objC"ish...
I vote for just deleting this typealias and declaring line 22 as follows:

` public var onStateChange: (() -> Void)?`